Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionRepository.kt

```diff
@@ -182,6 +182,7 @@ class SubmissionRepository @Inject constructor(
         LocalData.initialTestResultReceivedTimestamp(0L)
         LocalData.isAllowedToSubmitDiagnosisKeys(false)
         LocalData.isTestResultAvailableNotificationSent(false)
+        LocalData.numberOfSuccessfulSubmissions(0)
```

Comment:
There might be an issue with the state still.

* During positive test, key download should keep running for plausible deniability?
* After removing the positive test key download is only enabled after restarting the app

### Before first test
```bash
adb shell am broadcast -a "androidx.work.diagnostics.REQUEST_DIAGNOSTICS" -p de.rki.coronawarnapp.test

2021-02-08 19:37:17.065 21765-22027/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Recently completed work:
2021-02-08 19:37:17.070 21765-22027/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    e79c3dcb-2c78-40ab-9301-2bce816fbf60	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	 null	 SUCCEEDED	 DiagnosisKeyBackgroundOneTimeWork	 DIAGNOSIS_KEY_ONE_TIME_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	
    22838970-130e-4cb8-961f-f02d7cea3ea1	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	 null	 SUCCEEDED	 	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	
2021-02-08 19:37:17.070 21765-22027/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Running work:
2021-02-08 19:37:17.072 21765-22027/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    24a5b31f-37ee-4836-a9ae-f647f1543af2	 androidx.work.impl.workers.DiagnosticsWorker	 9	 RUNNING	 	 androidx.work.impl.workers.DiagnosticsWorker	
2021-02-08 19:37:17.072 21765-22027/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Enqueued work:
2021-02-08 19:37:17.080 21765-22027/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    0693b399-bf19-43a2-aea3-1e6006f0734f	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	 5	 ENQUEUED	 DiagnosisKeyBackgroundPeriodicWork	 DIAGNOSIS_KEY_PERIODIC_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	
    eae464f2-0163-4f3c-ba6a-8f371c43ab8c	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	 6	 ENQUEUED	 DeadmanNotificationOneTimeWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	
    e3ab1f79-8b33-4883-8287-e65c9663dcf1	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	 7	 ENQUEUED	 DeadmanNotificationPeriodicWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	
    ee7383df-587a-4bce-91ad-8cc3ea9be327	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	 8	 ENQUEUED	 ContactDiaryPeriodicWork	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	

```

### Positive test and submission
```bash
adb shell am broadcast -a "androidx.work.diagnostics.REQUEST_DIAGNOSTICS" -p de.rki.coronawarnapp.test

2021-02-08 19:38:58.558 21765-22170/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Recently completed work:
2021-02-08 19:38:58.573 21765-22170/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    ec4eb9da-0466-43b5-a8c4-a024f63df9de	 de.rki.coronawarnapp.submission.auto.SubmissionWorker	 null	 CANCELLED	 AutoSubmissionWorker	 AutoSubmissionWorker,de.rki.coronawarnapp.submission.auto.SubmissionWorker	
    24a5b31f-37ee-4836-a9ae-f647f1543af2	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    e3ab1f79-8b33-4883-8287-e65c9663dcf1	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	 null	 CANCELLED	 DeadmanNotificationPeriodicWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	
    eae464f2-0163-4f3c-ba6a-8f371c43ab8c	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	 null	 CANCELLED	 DeadmanNotificationOneTimeWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	
    0693b399-bf19-43a2-aea3-1e6006f0734f	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	 null	 CANCELLED	 DiagnosisKeyBackgroundPeriodicWork	 DIAGNOSIS_KEY_PERIODIC_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	
    e79c3dcb-2c78-40ab-9301-2bce816fbf60	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	 null	 SUCCEEDED	 DiagnosisKeyBackgroundOneTimeWork	 DIAGNOSIS_KEY_ONE_TIME_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	
    22838970-130e-4cb8-961f-f02d7cea3ea1	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	 null	 SUCCEEDED	 	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	
2021-02-08 19:38:58.573 21765-22170/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Running work:
2021-02-08 19:38:58.575 21765-22170/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    4af8a51a-a833-41ae-9b1e-9874e67399c7	 androidx.work.impl.workers.DiagnosticsWorker	 15	 RUNNING	 	 androidx.work.impl.workers.DiagnosticsWorker	
2021-02-08 19:38:58.575 21765-22170/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Enqueued work:
2021-02-08 19:38:58.578 21765-22170/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    01cc12e3-64f7-4b54-96a7-06e58ae52e8b	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	 12	 ENQUEUED	 ContactDiaryPeriodicWork	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	
```

### Test removed and tracing re-enabled
```bash
adb shell am broadcast -a "androidx.work.diagnostics.REQUEST_DIAGNOSTICS" -p de.rki.coronawarnapp.test

2021-02-08 19:40:16.510 22200-22331/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Recently completed work:
2021-02-08 19:40:16.536 22200-22331/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    40785e3c-aae7-4605-adc6-0cd4f571e915	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    8892a4df-0edb-45c2-b040-be70e1b45a93	 androidx.work.impl.workers.DiagnosticsWorker	 16	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    4af8a51a-a833-41ae-9b1e-9874e67399c7	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    ec4eb9da-0466-43b5-a8c4-a024f63df9de	 de.rki.coronawarnapp.submission.auto.SubmissionWorker	 null	 CANCELLED	 AutoSubmissionWorker	 AutoSubmissionWorker,de.rki.coronawarnapp.submission.auto.SubmissionWorker	
    24a5b31f-37ee-4836-a9ae-f647f1543af2	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    eae464f2-0163-4f3c-ba6a-8f371c43ab8c	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	 null	 CANCELLED	 DeadmanNotificationOneTimeWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	
    0693b399-bf19-43a2-aea3-1e6006f0734f	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	 null	 CANCELLED	 DiagnosisKeyBackgroundPeriodicWork	 DIAGNOSIS_KEY_PERIODIC_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	
    e79c3dcb-2c78-40ab-9301-2bce816fbf60	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	 null	 SUCCEEDED	 DiagnosisKeyBackgroundOneTimeWork	 DIAGNOSIS_KEY_ONE_TIME_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	
    22838970-130e-4cb8-961f-f02d7cea3ea1	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	 null	 SUCCEEDED	 	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	
2021-02-08 19:40:16.536 22200-22331/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Running work:
2021-02-08 19:40:16.538 22200-22331/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    fdc2204f-6c4d-45f5-94b8-7ae412fd2a54	 androidx.work.impl.workers.DiagnosticsWorker	 21	 RUNNING	 	 androidx.work.impl.workers.DiagnosticsWorker	
2021-02-08 19:40:16.538 22200-22331/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Enqueued work:
2021-02-08 19:40:16.543 22200-22331/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    d97ceb3a-bd2c-49b2-8222-b60f64bffc69	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	 18	 ENQUEUED	 ContactDiaryPeriodicWork	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	
    284e1d7b-ca04-412e-bbee-4b0a939b500c	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	 20	 ENQUEUED	 DeadmanNotificationPeriodicWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	
```

### Test removed, tracing re-enabled and app restarted
```bash
adb shell am broadcast -a "androidx.work.diagnostics.REQUEST_DIAGNOSTICS" -p de.rki.coronawarnapp.test

2021-02-08 19:41:16.755 22379-22521/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Recently completed work:
2021-02-08 19:41:16.784 22379-22521/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    f5f3036d-2c32-4d59-9511-fb2038b5883c	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    fdc2204f-6c4d-45f5-94b8-7ae412fd2a54	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    40785e3c-aae7-4605-adc6-0cd4f571e915	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    8892a4df-0edb-45c2-b040-be70e1b45a93	 androidx.work.impl.workers.DiagnosticsWorker	 16	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    4af8a51a-a833-41ae-9b1e-9874e67399c7	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    ec4eb9da-0466-43b5-a8c4-a024f63df9de	 de.rki.coronawarnapp.submission.auto.SubmissionWorker	 null	 CANCELLED	 AutoSubmissionWorker	 AutoSubmissionWorker,de.rki.coronawarnapp.submission.auto.SubmissionWorker	
    24a5b31f-37ee-4836-a9ae-f647f1543af2	 androidx.work.impl.workers.DiagnosticsWorker	 null	 SUCCEEDED	 	 androidx.work.impl.workers.DiagnosticsWorker	
    eae464f2-0163-4f3c-ba6a-8f371c43ab8c	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	 null	 CANCELLED	 DeadmanNotificationOneTimeWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationOneTimeWorker	
    e79c3dcb-2c78-40ab-9301-2bce816fbf60	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	 null	 SUCCEEDED	 DiagnosisKeyBackgroundOneTimeWork	 DIAGNOSIS_KEY_ONE_TIME_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalOneTimeWorker	
    22838970-130e-4cb8-961f-f02d7cea3ea1	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	 null	 SUCCEEDED	 	 de.rki.coronawarnapp.nearby.ExposureStateUpdateWorker	
2021-02-08 19:41:16.784 22379-22521/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Running work:
2021-02-08 19:41:16.786 22379-22521/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    5ed01f56-e6f9-4845-8d02-42d815460523	 androidx.work.impl.workers.DiagnosticsWorker	 27	 RUNNING	 	 androidx.work.impl.workers.DiagnosticsWorker	
2021-02-08 19:41:16.786 22379-22521/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr: Enqueued work:
2021-02-08 19:41:16.793 22379-22521/de.rki.coronawarnapp.test I/WM-DiagnosticsWrkr:  Id 	 Class Name	 Job Id	 State	 Unique Name	 Tags	
    284e1d7b-ca04-412e-bbee-4b0a939b500c	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	 20	 ENQUEUED	 DeadmanNotificationPeriodicWork	 de.rki.coronawarnapp.deadman.DeadmanNotificationPeriodicWorker	
    23d6eb6b-4d9f-49c4-a245-6b43b613399c	 de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	 24	 ENQUEUED	 DiagnosisKeyBackgroundPeriodicWork	 DIAGNOSIS_KEY_PERIODIC_WORKER,de.rki.coronawarnapp.worker.DiagnosisKeyRetrievalPeriodicWorker	
    3e2276a6-a220-4788-9294-cad09c7e286f	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	 26	 ENQUEUED	 ContactDiaryPeriodicWork	 de.rki.coronawarnapp.contactdiary.retention.ContactDiaryRetentionWorker	
```

Reply:
Yes, the key download should be running all the time. It's fixed in the future commit.